src/storage/src/hummock/sstable/mod.rs

```diff
@@ -485,18 +500,18 @@ impl SstableMeta {
             monotonic_tombstone_events.push(monotonic_tombstone_event);
         }
         let meta_offset = buf.get_u64_le();
-
-        Ok(Self {
+        let meta = Self {
             block_metas,
-            bloom_filter,
+            bloom_filter: vec![],
```

Comment:
Though setting a `bloom_filter` to empty here is correct because `Sstable::with_filter_reader` always comes after `SstableMeta::decode`, I think it is dangerous to do so and prone to future bugs (imagine someone thinks the `bloom_filter` field is valid all the time and try to re-use it).

I think it is cleaner to replace `pub bloom_filter: Vec<u8>` to  [enum XorFilter](https://github.com/risingwavelabs/risingwave/blob/398d96eaea9da5d267b6913c0f0babb18d2cf37a/src/storage/src/hummock/sstable/xor_filter.rs#L277). For write path, we can build `Vec<u8>` from `XorFilter`. For read path, we can build `XorFilterReader` from `XorFilter`.